crates/lint/src/sol/mod.rs

```diff
@@ -257,6 +263,37 @@ impl<'a> Linter for SolidityLinter<'a> {
                 }
             });
         });
+
+        // Handle diagnostics and fail if necessary.
+        let s = |count| if count == 1 { "" } else { "s" };
```

Comment:
nit: IMO would be OK to just display
```
                "aborting due to 1 linter warning(s) ",
```

```
                "1 linter warning(s) and 1 note(s)",
```

and avoid extra logic to count / format, but happy to keep it this way too